cli/command/trust/revoke.go

```diff
@@ -1,17 +1,137 @@
 package trust
 
 import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/docker/notary/client"
+	"github.com/docker/notary/tuf/data"
+
 	"github.com/docker/cli/cli"
 	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/trust"
+	"github.com/docker/distribution/reference"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/registry"
 	"github.com/spf13/cobra"
 )
 
 func newRevokeCommand(dockerCli command.Cli) *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "revoke [OPTIONS]",
+		Use:   "revoke [OPTIONS] IMAGE[:TAG]",
 		Short: "Remove trust for an image",
-		Args:  cli.NoArgs,
-		RunE:  command.ShowHelp(dockerCli.Err()),
+		Args:  cli.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return revokeTrust(dockerCli, args[0])
+		},
 	}
 	return cmd
 }
+
+func askConfirm(input io.Reader) bool {
+	var res string
+	if _, err := fmt.Fscanln(input, &res); err != nil {
+		return false
+	}
+	if strings.EqualFold(res, "y") || strings.EqualFold(res, "yes") {
+		return true
+	}
+	return false
+}
+
+func revokeTrust(cli command.Cli, remote string) error {
+	ref, repoInfo, authConfig, err := getImageReferencesAndAuth(cli, remote)
+	if err != nil {
+		return err
+	}
+	switch ref.(type) {
+	case reference.Digested, reference.Canonical:
+		return fmt.Errorf("Cannot remove signature for digest")
+	case reference.NamedTagged:
+		if err := revokeSingleSig(cli, ref.(reference.NamedTagged), repoInfo, *authConfig); err != nil {
+			return fmt.Errorf("Could not remove signature")
+		}
+		return nil
+	default:
+		if err := revokeAllSigs(cli, ref, repoInfo, *authConfig); err != nil {
+			return fmt.Errorf("Could not remove all signatures")
```

Comment:
nit: would be nice to include the original user-specified repo name in this error, and if any detail from the error itself is useful.  Also do we need a newline?